src/pages/index.js

```diff
@@ -29,54 +32,153 @@ const enableValidation = (config) => {
 
 enableValidation(config);
 
-const createCard = (cardItem) => {
-  const card = new Card(cardItem, "#card-template", () => {
-    console.log(cardItem);
-    imgPopup.open(cardItem);
+//API
+
+export const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authorization: "7dcd9a93-149c-4e56-87db-9285c9177a9e",
+});
+
+api
+  .getUserData()
+  .then((results) => results.json())
+  .then((data) => {
+    data;
+    currentAvatar = data.avatar;
+    initialUserData = new UserInfo(data);
   });
+
+api.getInitialCards().then((data) => {
+  data;
+
+  cardSection = new Section(
+    {
+      data: data,
+      renderer: (data) => {
+        const cardElement = createCard(data);
+        cardSection.addItem(cardElement);
+      },
+    },
+    cardListElement
+  );
+  cardSection.renderItems();
+});
+
+const createCard = (cardItem) => {
+  const card = new Card(
+    cardItem,
+    "#card-template",
+    () => {
+      imgPopup.open(cardItem);
+    },
+    () => {
+      deleteConfirmPopup.open(card);
+    }
+  );
   return card.getView();
 };
 
+//VAR
+
+let cardSection;
+let initialUserData;
+let currentAvatar;
+let profileData;
+let newCardData;
+
+//Popups
+
 const imgPopup = new PopupWithImages("#card-picture-modal");
 imgPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    data: initialCards,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-      cardSection.addItem(cardElement);
-    },
-  },
-  cardListElement
+const profilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileUpdate
 );
-cardSection.renderItems();
-
-const userInfo = new UserInfo(".profile__name", ".profile__job");
-
-const profilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
-  profilePopup.close();
-  userInfo.setUserInfo(data);
-});
 profilePopup.setEventListeners();
 
-const newCardPopup = new PopupWithForm("#card-add-modal", (data) => {
-  console.log(data);
-  const cardFormElement = createCard(data);
-  cardSection.addItem(cardFormElement);
-  newCardPopup.close();
-});
+const newCardPopup = new PopupWithForm("#card-add-modal", handleAddCard);
 newCardPopup.setEventListeners();
 
+export const deleteConfirmPopup = new PopupConfirmation(
+  "#modal-delete",
+  handleDeleteConfirmation
+);
+deleteConfirmPopup.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#modal-avatar", handleUpdateAvatar);
+avatarPopup.setEventListeners();
+
+//HANDLES
+
+function handleDeleteConfirmation(card, id) {
+  card.handleDeleteCard(id);
+}
+
+function handleProfileUpdate(userData) {
+  initialUserData.setUserInfo(userData);
```

Comment:
You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change

**Please, correct it everywhere**

Reply:
Corrected everywhere